### ISSUE TITLE
LevelDB Connection Crash

### ISSUE THREAD

**Issue by sqwk** [protonail/keylord-issues-and-feature-requests#31]
When connecting to a LevelDB database, the app crashes. Running 4.0.2 on Mac OS 10.13.3.

    default	14:08:54.820353 +0100	kernel	AGC:: [Keylord pid 90796 non-mux-aware] exiting, non-mux-aware app count 2, runtime: 0:00:05.028
    default	14:08:54.820523 +0100	kernel	AGC:: [Keylord pid 90796 mux-aware] exiting, non-mux-aware app count 2, runtime: 0:00:05.033

**Comment by maxd** [comment]
Could you please provide crash-logs files of Keylord application? You can find them in "Console" application into "User Reports" section:

![crashlog](https://user-images.githubusercontent.com/27208/37522493-229cb764-2935-11e8-8879-44a9e90b57b5.png)

**Please, don't insert them to the comment. Just attache them as file.**

**Comment by sqwk** [comment]
Oddly enough, there isn't one. However, system.log lists this:

    Mar 16 14:24:11 Kiwi Keylord[94137]: Keylord(94137,0x700009921000) malloc: *** error for object 0x1334d6000: pointer being freed was not allocated
	    *** set a breakpoint in malloc_error_break to debug
    Mar 16 14:24:13 Kiwi com.apple.xpc.launchd[1] (com.apple.ReportCrash): Service only ran for 1 seconds. Pushing respawn out by 9 seconds.
    Mar 16 14:24:13 Kiwi com.apple.xpc.launchd[1] (com.apple.ReportCrash.Root): Service only ran for 1 seconds. Pushing respawn out by 9 seconds.

**Comment by maxd** [comment]
Yes, it is very strange. It looks like the google/leveldb#199 problem in LevelDB database.

How do you think can you create test database for reproduce problem?

**Comment by sqwk** [comment]
Same problem when creating a db through Keylord. (See attached test db)
[test.zip](https://github.com/protonail/keylord-issues-and-feature-requests/files/1819426/test.zip)

**Comment by maxd** [comment]
I can open `test` database use Keylord 4.0.2 downloaded from site without any problems. It display empty database.

Looks like this is environment specific problem. I will try to update my Mac OS up to 10.13.3 and try to reproduce problem.

For any case could you please reboot OS and try to reproduce problem?

**Comment by sqwk** [comment]
A restart did not help. I also tried clearing system and user caches.

**Comment by maxd** [comment]
Update of Mac OS did't help too. Keylord can open `test` database without problems.

I am going to provide development version of Keylord next week. It is using latest version of LevelDB adapter. Maybe it fixed the problem.

**Comment by sqwk** [comment]
Sounds good. Alternatively, do you have a download link for an older version that I could test?

**Comment by maxd** [comment]
Let's try to check latest version first because I don't think that old version will work better. 

Here you can download development version of Keylord application: https://protonail.com/keylord/download/development

**Comment by sqwk** [comment]
Still the same issue. The app quits without any errors or a crash log.

**Comment by maxd** [comment]
I have simulated unhandled exception (not related to this problem) and in my case the OS show the following dialog:

![unhandled-exception-dialog](https://user-images.githubusercontent.com/27208/37790210-ebe06770-2e16-11e8-81e7-fc6ae5b33e1b.png)

It have "Report..." button which can open other dialog with detail information about problem (after click by "Show Details" button).

Do you see such dialog after application crash?

**Comment by sqwk** [comment]
No, unfortunately there is no dialogue. It almost looks like the app simply quits, although this was not triggered by me. Not sure what could be causing this. I have SIP disabled and am using the NVIDIA WebDriver, but I don't think that should affect the app in any way. Also FileVault is enabled.

**Comment by sqwk** [comment]
All tests ran successfully. I think I will wait for the next OS update and see if that changes anything. Still no logs or crash reports.
